test/poll.go

```diff
@@ -0,0 +1,20 @@
+package test
+
+import (
+	"testing"
+	"time"
+)
+
+// Poll repeatedly evaluates condition until we either timeout, or it suceeds.
+func Poll(t *testing.T, d time.Duration, condition func() bool, msg string) {
+	deadline := time.Now().Add(d)
+	for {
+		if time.Now().After(deadline) {
+			t.Fatal(msg)
+		}
+		if condition() {
+			return
+		}
+		time.Sleep(d / 10)
```

Comment:
Why  `d / 10`? Wouldn't it make sense to make it configurable (or a fixed amount)?